### PR TITLE
tools/rbd, rgw: Removed unreachable returns

### PR DESCRIPTION
**Pull request by joscollin**
Removed unreachable returns as the usage() always exits the process. This PR is created as a
follow up of the review in: https://github.com/ceph/ceph/pull/16254.

Signed-off-by: Jos Collin <jcollin@redhat.com>

### CODE REVIEW

**Comment by joscollin** [comment]
Jenkins Retest this please

**Comment by joscollin** [comment]
Jenkins Retest this please

**Comment by tchaikov** [comment]
retest this please.

**Comment by joscollin** [comment]
Jenkins Retest this please

File: src/tools/rbd_nbd/rbd-nbd.cc [hunk 1]
```diff
@@ -1061,7 +1060,7 @@ static int rbd_nbd(int argc, const char *argv[])
       break;
     default:
       usage();
-      return -EINVAL;
```
Comment (dillaman) [inline src/tools/rbd_nbd/rbd-nbd.cc]: Why doesn't this return failure anymore?

File: src/rgw/rgw_admin.cc [hunk 2]
```diff
@@ -2410,7 +2409,6 @@ int main(int argc, const char **argv)
       break;
     } else if (ceph_argparse_flag(args, i, "-h", "--help", (char*)NULL)) {
       usage();
-      return 0;
```
Comment (dillaman) [inline src/rgw/rgw_admin.cc]: Shouldn't these return statements be kept to actually stop further processing in this method?

**Comment by joscollin** [comment]
@dillaman Thanks for reviewing. I have removed those returns, as usage() always executes `exit(1)` for sure. I have specified the same in the PR description.

**Comment by dillaman** [comment]
@joscollin Probably needs to be updated then since ```usage()``` does not directly invoke ```exit(1)``` -- and if ```_usage()``` does, it might be good to add an ```assert(false)``` after the call to ensure it's never removed in the future.

**Comment by joscollin** [comment]
@dillaman Fixed as per your suggestion. Could you please review ? Thanks.

**Comment by joscollin** [comment]
@dillaman ping

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2017-08-01_19:45:33-rgw-wip-yuri-testing3_2017_8_1_2-distro-basic-smithi/

**Pull request merged**